flasksrc/quiz.py

```diff
@@ -0,0 +1,43 @@
+from flask import Flask, request
+from flask_restful import Resource, Api, abort
+
+from flasksrc.db import get_db
+
+class ToyNetQuiz(Resource):
+    def get(self, quiz_id):
+        db = get_db()
+        error = None
+
+        rows_question = db.execute(
+            'SELECT qq.question_id, qq.question, qq.answer'
+            ' FROM toynet_quiz_questions AS qq'
+            ' WHERE qq.quiz_id = (?) ',
+            (str(quiz_id),)
+        ).fetchall()
+
+        rows_option = db.execute(
```

Comment:
Style Question: Any reason why you don't do a join operation here?

Reply:
When I first did the join operation to `toynet_quiz_questions` and `toynet_quiz_options`, an `<sqlite3.Row object>` was made for each option in the question.
The resulted json was this:
```
[
    {
        "question": "Which type of fiber is best suited for long distance transmission?",
        "options": [
            "Multi-Mode Fiber"
        ],
        "answer": 1
    },
    {
        "question": "Which type of fiber is best suited for long distance transmission?",
        "options": [
            "Single-mode Fiber"
        ],
        "answer": 1
    },
    {
        "question": "Which type of fiber is best suited for long distance transmission?",
        "options": [
            "Small Form Factor Pluggable"
        ],
        "answer": 1
    },
    {
        "question": "Which type of fiber is best suited for long distance transmission?",
        "options": [
            "Fast Fiber"
        ],
        "answer": 1
    },
```
instead of:
```
[
    {
        "question": "Which type of fiber is best suited for long distance transmission?",
        "options": [
            "Multi-Mode Fiber",
            "Single-mode Fiber",
            "Small Form Factor Pluggable",
            "Fast Fiber"
        ],
        "answer": 1
    },
```
I first tried to fix this using SQL, but using python was far simpler

Reply:
You can probably get a more elegant solution doing something like the [this `LEFT JOIN`](https://github.com/Project-Reclass/toynet-flask/blob/cffd539bed1343c3ed95cd9b094a91bb1b1747a5/flasksrc/value.py#L14) and [list comprehension](https://github.com/Project-Reclass/toynet-flask/blob/cffd539bed1343c3ed95cd9b094a91bb1b1747a5/flasksrc/value.py#L25) from [`value.py`](https://github.com/Project-Reclass/toynet-flask/blob/main/flasksrc/value.py).

Reply:
If I LEFT JOIN the two tables, the values for question_id, question, and answer are repeated for every option. <sqlite3.Row object> is made for each option in the question.
How do I make the wanted json structure with such a join table? I can do it in an awkward way, but is there a better way?

Reply:
It is a bit more complex than just a list comprehension, here is something that should work:

```
rows_question = db.execute(
    'SELECT q.question_id, q.question, q.answer, qo.option'
    ' FROM toynet_quizzes AS q'
    ' LEFT JOIN toynet_quiz_options as qo'
    ' on q.quiz_id = qo.quiz_id AND q.question_id = qo.question_id'
    ' WHERE q.quiz_id = (?) '
    ' ORDER BY q.question_id, qo.option_id',
    (str(quiz_id),)
).fetchall()
```
And to reconcile the duplicates:

```
res = list()
cur_quiz_id = None

for row in rows_question:
    if cur_quiz_id != row['question_id']:
        cur_quiz_id = row['question_id']
        res.append({
            'question': row['question'],
            'options': [row['option']],
            'answer': row['answer']})
    else:
        res[-1]['options'].append(row['option'])
 return res, 200
```

You would still need to do your sanity check to ensure that the answer is <= the length of the options.